src/window.rs

```diff
@@ -247,6 +260,16 @@ impl Window {
                 }
             }
             WindowEvent::MouseInput { state, button, .. } => {
+                // handle context menu
+                // TODO: should create on ShowContextMenu event
+                if *button == winit::event::MouseButton::Right && *state == ElementState::Pressed {
+                    self.show_context_menu();
+                    // FIXME: there's chance to lose the event since the channel is async.
+                    if let Ok(event) = self.menu_event_receiver.try_recv() {
```

Comment:
Is this the reason that could get blocking on rendering. I saw some weird behavior might because of this.

Reply:
I think it is. But It will be complicated if we make it asynchronous because we'll need to find a loop to keep listening to the menu event channel, and we can't tell if the menu event is from which context menu since it only returns the option you select, so there will be a chance we act on a wrong window.

Also, we'll create the menu item every time for a different context. If we do it asynchronously in a loop somewhere, we'll lose the context since the menu event channel only passes the item ID.

For example, Servo passes a `ShowContextMenu` event with the text "AAA" selected. We should create a context with options to handle the text like `Copy`. Once it becomes asynchronous and executes outside the context, we'll only know the user selects `Copy`, but we don't know what text to copy.